src/Reflection/ClassReflection.php

```diff
@@ -672,11 +672,24 @@ public function getInterfaces(): array
 	/**
 	 * @return \PHPStan\Reflection\ClassReflection[]
 	 */
-	public function getTraits(): array
+	public function getTraits(bool $recursive = false): array
 	{
-		return array_map(function (\ReflectionClass $trait): ClassReflection {
+		$traits = array_map(function (\ReflectionClass $trait): ClassReflection {
 			return $this->reflectionProvider->getClass($trait->getName());
-		}, $this->getNativeReflection()->getTraits());
+		}, $this->collectTraits($this->getNativeReflection()));
```

Comment:
This needs to stay `$this->getNativeReflection()->getTraits()` as before. Only with recursive=true we are allowed to also return traits from nested traits and parent classes.

Reply:
If I change this back my new tests are failing as the nested traits are not discovered...

There are now two changes in `getTraits`:
- `$this->collectTraits($this->getNativeReflection())` instead of `$this->getNativeReflection()->getTraits()`
    - This allows nested traits to be discovered
-   Added `bool $recursive = false`
    - This allows traits from parent classes to be discovered